components/organisation-unit-tree/src/organisation-unit-tree/default-render-node-label/label/label.js

```diff
@@ -0,0 +1,118 @@
+import propTypes from 'prop-types'
+import React from 'react'
+import { orgUnitPathPropType } from '../../../prop-types.js'
+import { DisabledSelectionLabel } from './disabled-selection-label.js'
+import { IconizedCheckbox } from './iconized-checkbox.js'
+import { LabelContainer } from './label-container.js'
+import { SingleSelectionLabel } from './single-selection-label.js'
+
+const createNewSelected = ({ selected, path, checked, singleSelection }) => {
+    const pathIndex = selected.indexOf(path)
+
+    if (checked && pathIndex !== -1) return selected
+    if (singleSelection && checked) return [path]
+    if (checked) return [...selected, path]
+    if (pathIndex === -1) return selected
+    if (singleSelection) return []
+    if (selected.indexOf(path) === 0) return selected.slice(1)
+
+    const prevSlice = selected.slice(0, pathIndex)
+    const nextSlice = selected.slice(pathIndex + 1)
+    return [...prevSlice, ...nextSlice]
+}
+
+const Label = ({
+    id,
+    path,
+    open,
+    loading,
+    checked,
+    onChange,
+    dataTest,
+    selected,
+    hasChildren,
+    highlighted,
+    displayName,
+    onToggleOpen,
+    disableSelection,
+    singleSelection,
+    hasSelectedDescendants,
+}) => {
+    const onClick = ({ checked }, event) => {
+        const newSelected = createNewSelected({
+            selected,
+            path,
+            checked,
+            singleSelection,
+        })
+
+        onChange(
+            { id, path, checked, displayName, selected: newSelected },
+            event
+        )
+    }
+
+    if (disableSelection) {
+        return (
+            <LabelContainer highlighted={highlighted}>
+                <DisabledSelectionLabel
+                    label={displayName}
+                    loading={loading}
+                    onToggleOpen={onToggleOpen}
+                />
+            </LabelContainer>
+        )
+    }
+
+    if (singleSelection) {
+        return (
+            <LabelContainer highlighted={highlighted}>
+                <SingleSelectionLabel
+                    checked={checked}
+                    label={displayName}
+                    onChange={onClick}
+                    loading={loading}
+                >
+                    {displayName}
```

Comment:
[SingleSelectionLabel](https://github.com/dhis2/ui/blob/1712a034f9acc5e287e07d1aca452e0452dac5c8/components/organisation-unit-tree/src/organisation-unit-tree/default-render-node-label/label/single-selection-label.js) does not use the `children` prop so I think this can just be

```jsx
                <SingleSelectionLabel
                    checked={checked}
                    label={displayName}
                    onChange={onClick}
                    loading={loading}
                />
```